File: Java/Palindrome Number.java
// 10ms, 25.42%; 40.5MB, 5.02%
class Solution {
    public boolean isPalindrome(int x) {
        StringBuilder s1 = new StringBuilder(String.valueOf(x));
        StringBuilder s2 = new StringBuilder(s1).reverse();
        return s1.compareTo(s2) == 0;
    }
}

File: Go/Palindrome Number.go
// 4ms, 98.98%; 5.4MB, 25.00%
func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}
	str := strconv.Itoa(x)
	length := len(str)
	for i := length >> 1; i > -1; i-- {
		if str[i] != str[length-i-1] {
			return false
		}
	}
	return true
}
